.circleci/config.yml

```diff
@@ -114,6 +114,10 @@ jobs:
           name: Run typescript tests
           command: |
             python -m scripts.typescript_checks
+      - run:
+          name: Run typescript tests in strict mode
+          command: |
+            python -m scripts.typescript_checks --strict_checks
```

Comment:
My understanding is that we want the typescript checks to pass before running e2e tests, that's why we have these in CircleCI. Is strict check tests required to pass for the e2e tests to run? Or is it specific to typescript checks itself? 
If its the latter, can we move it to Github Actions? @nithusha21 FYI

Reply:
I don't know, actually do we need both typescript checks and the strict typescript checks? What's the difference? Would just the strict checks suffice?

Reply:
Actually the thing is that we want to make the typescript checks more strict so we are enabling the `strict` flag in tsconfig. But the strict checks can't be enabled in the complete codebase due to a lot of ts errors. You can refer the discussion at #10327 

So, only some files are compiled using the strict config and we plan to cover the complete codebase gradually.

I think we can enable the strict checks in circle ci given that they take a less amount of time (currently only 3s) and the checks are also present in pre_push_hoooks.

Reply:
Ah okay, sounds good. Thanks for the explanation!